web/modules/custom/asu_admin_toolbox/src/Plugin/Action/SolrReindexChildAction.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace Drupal\asu_admin_toolbox\Plugin\Action;
+
+use Drupal\Core\Action\ActionBase;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Session\AccountInterface;
+use Psr\Log\LoggerInterface;
+use Drupal\Core\Entity\EntityTypeManager;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Url;
+use Symfony\Component\Process\Process;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+use Drupal\asu_islandora_utils\AsuUtils;
+
+/**
+ * Reindex an item and children in Solr.
+ *
+ * @Action(
+ *   id = "solr_reindex_children",
+ *   label = @Translation("Solr Reindex Item and Children"),
+ *   type = "node"
+ * )
+ */
+class SolrReindexChildAction extends ActionBase implements ContainerFactoryPluginInterface
+{
+
+  /**
+   * The entityTypeManager definition.
+   *
+   * @var Drupal\Core\Entity\EntityTypeManager
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Logger.
+   *
+   * @var Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * The AsuUtils definition.
+   *
+   * @var \Drupal\asu_islandora_utils\AsuUtils
+   */
+  protected $asuUtils;
+
+  /**
+   * Constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param Drupal\Core\Entity\EntityTypeManager $entityTypeManager
+   *   The entityTypeManager definition.
+   * @param Psr\Log\LoggerInterface $logger
+   *   Logger.
+   * @param $ASUUtils
+   *   The ASU Utils service.
+   */
+  public function __construct(
+      array $configuration,
+      $plugin_id,
+      $plugin_definition,
+      EntityTypeManager $entityTypeManager,
+      LoggerInterface $logger,
+      AsuUtils $ASUUtils
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityTypeManager = $entityTypeManager;
+    $this->logger = $logger;
+    $this->asuUtils = $ASUUtils;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('logger.channel.islandora'),
+      $container->get('asu_utils')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute($entity = NULL) {
+    if (!$entity) {
+      return;
+    }
+    $entity_type = $entity->getEntityTypeId();
+
+    if (!$entity_type == 'node') {
+      return;
+    }
+
+    $content_type = $entity->bundle();
+    if ($entity->getEntityTypeId() == 'node' && $content_type == 'asu_repository_item') {
+      \Drupal::logger('solr reindex item and children action')->info("about to reindex item and children");
```

Comment:
same as above - i'm not sure we need to log here, but if you think it's helpful to have in production then i'd say it should at least include the parent item's node id